hack/unit.sh

```diff
@@ -13,6 +13,5 @@ if [[ "$goflags" == *"-mod=vendor"* ]]; then
   unset GOFLAGS
 fi
 
-go test -v ./cmd/...
-
+go test -v `go list ./... | grep -v test/e2e`
```

Comment:
Not sure what is going on here. Please add a comment to explain.

Reply:
Having just `go test ./..` , also runs e2e tests. For unit tests, we need to exclude `e2e` folder, will add this in the comment

Reply:
It's like listing all test files in all the directories at the top excluding `e2e`, will it have any side effect if we have an integration test in future :P ? We need to ensure that only unit tests exist in all the packages.

Reply:
I suggest we leave it as it is with a comment and revisit this when introducing integration tests unless you have a better way of doing. If yes, please specify that.

Reply:
We can either use `short` (https://golang.org/pkg/testing/#Short) while writing integration tests or use `build tags`.  @ravisantoshgudimetla Which method should we follow ?

Reply:
Nope, short is to indicate that certain tests are long running that we want to avoid those kind of tests. I believe you're talking about `build tags` in general which ensure that certain packages are not built at all which is not what we want.

I think go community wanted explicit mentioning of packages. It also is related to the way kube/OpenShift/Operator-sdk have put tests, we always have `test/e2e` to indicate e2e tests. So, what's wrong with `go test -v ./cmd/... ./pkg/...` also add a `count=1` so that we will be explicit and to not re-use the cache?

Reply:
@VaishnaviHire I might have sent you down a wrong path by asking you to specify `./`. Sorry about that.

Reply:
Done. @aravindhp No worries. The only downside I feel, is requiring to explicitly mention the packages every time we add tests

Reply:
Why would we have to explicitly mention packages? `./cmd/... ./pkg/...` should cover everything. I do see us adding new packages outside of `/pkg`.